src/main/io/osd.c

```diff
@@ -4239,57 +4364,60 @@ static void osdRefresh(timeUs_t currentTimeUs)
 #endif
             osdSetNextRefreshIn(delay);
         } else {
-            osdShowStatsPage1(); // show first page of statistics
+            osdShowStats(osdVideoSystemIsSinglePageStatsCompatible(), 1); // show first page of statistics
```

Comment:
Just a thought. Instead of having that function, you could just use this
`osdShowStats((osdDisplayPort->rows >= 18), 1);`
Though this won't explicitly check for HD video systems, it will make sure that the height of the OSD is adequate. Even with the minimum ratio of 4:3, that would be 32 characters wide. But currently there are no 4:3 HD displays at 18 characters high. So 50 characters max width would be fine. Even HDZero in 4:3 mode uses the full 16:9 for the stats screen.

Reply:
I like that that. It makes sense and would eliminate the need to modify this again to accommodate Avatar.  I'll go ahead and do that.  I may add a constant to hold that 18 value. Something like MAX_LINES_FOR_SINGLE_PAGE or whatever.

Reply:
I made the changes you suggested. Detection of compatablility for single-page stats is now based strictly on the number of rows currently displayed. OSD_STATS_SINGLE_PAGE_MIN_ROWS stores the minimum number of rows for single-page stats.

I also made some additional improvements surrounding the logic for manually paging using the stick commands for multi-page stats. I think this makes the code a lot easier to follow. 

There is one caveat surrounding the timing of when "Saved Settings" is displayed. Since it is only displayed when osdShowStats() is called, it could be missed if the 'stats_page_auto_swap_time' is 0, greater than 4 seconds or if the user has used the stick commands to manually page with multi-page stats since it effectively pauses the updates. So multi-page users would only see "Saved Settings" in that case if they happen to manually go to the next page within the 5 second window. Probably no big deal, but I figured I'd mention that. I also added this in the comments in the code for clarity.